src/app/app.component.ts

```diff
@@ -22,17 +34,35 @@ export class AppComponent implements OnInit {
     } else {
       this.router.navigate(['/login']);
     }
+
+    this._subscriptions.push(
+      this.loaderService.loading.subscribe((value: boolean) => {
+        this.loading = value;
+        this.changeDetectorRef.markForCheck();
+      })
+    );
+  }
+
+  ngOnDestroy(): void {
+    this._subscriptions.forEach((s: Subscription) => {
+      s.unsubscribe();
+    });
   }
 
   private getUserInfo(token: string): void {
-    this.authService.getUserInfo(token).subscribe(
-      (value: IUserInfo) => {
-        this.authService.authenticate();
-        this.router.navigate(['/courses']);
-      },
-      (error: HttpErrorResponse) => {
-        this.router.navigate(['/unauthorized']);
-      }
-    );
+    this.loaderService.loading.next(true);
+    this.authService
+      .getUserInfo(token)
+      .pipe(finalize((): void => this.loaderService.loading.next(false)))
```

Comment:
It would be better to move logic related to a loader to the http interceptor
https://medium.com/@zeljkoradic/loader-bar-on-every-http-request-in-angular-6-60d8572a21a9